com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs

```diff
@@ -137,7 +165,7 @@ public IPolicy GeneratePolicy(Func<float[]> heuristic)
         /// <param name="newBehaviorName">New name for the behavior.</param>
         /// <param name="model">New neural network model for this behavior.</param>
         /// <param name="inferenceDevice">New inference device for this behavior.</param>
-        public void GiveModel(
+        internal void SetModel(
```

Comment:
I think we want this internal and users should go through the Agent instead. Should we just get rid of this and have internal setters instead?

Note I renamed to SetModel in anticipation of https://github.com/Unity-Technologies/ml-agents/pull/3557

Reply:
I would be okay with internal setters yes.

Reply:
Done